src/PSR7/SpecFinder.php

```diff
@@ -52,12 +66,129 @@ public function findPathSpec(PathAddress $addr) : PathItem
         return $pathSpec;
     }
 
+    /**
+     * @return Parameter[]
+     *
+     * @throws NoPath
+     */
+    public function findPathSpecs(OperationAddress $addr) : array
+    {
+        $spec = $this->findOperationSpec($addr);
```

Comment:
I moved spec extracting to this class as is (with comments, etc). I think this class suits for such methods